build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,222 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Utility function to check expected failure for different platforms.
+function(platform_xfail OUTPUT_XFAIL XFAIL_ARGS)
+  set(${OUTPUT_XFAIL} FALSE PARENT_SCOPE)
+  if(XFAIL_ARGS)
+    if((CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+        CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+        XFAIL_ARGS MATCHES "rv64") OR
+       (ANDROID AND XFAIL_ARGS MATCHES "arm64-v8a") OR
+       (XFAIL_ARGS MATCHES "x86_64"))
+      set(${OUTPUT_XFAIL} TRUE PARENT_SCOPE)
+    endif()
+  endif()
+endfunction()
+
+
+# iree_run_module_test()
+#
+# Creates a test using iree-run-module to run an IREE module (vmfb).
+# The function supports cross compile, so there is no matching bzl rule.
+#
+# Parameters:
+#   NAME: Name of the target
+#   MODULE_SRC: IREE module (vmfb) file.
+#   BENCHMARK_MODULE_SRC: IREE module from benchmark_suite.
+#   DRIVER: Driver to run the module with. This can be overridden if it is
+#       provided by the benchmark_suite's flagfile.
+#   RUNNER_ARGS: additional args to pass to iree-run-module. The driver
+#       and input file are passed automatically.
+#   EXPECTED_OUTPUT: A file of expected output to compare with the output from
+#       iree-run-module
+#   LABELS: Additional labels to apply to the test. The package path and
+#       "driver=${DRIVER}" are added automatically.
+#   XFAIL: When set for x86_64|arm64-v8a|rv64, the test is expected to
+#       fail for the particular platforms/architectures for various reasons.
+#       e.g., benchmark_suite module is not available, upstream llvm backend
```

Comment:
I think we should keep `XFAIL` to mean running the test and having it fail, not something like the input files not being available where we can't even run the test. In the latter case, we should skip it altogether

Reply:
Yeah I'm confused by how this XFAIL property is being used. It seems sort of like a "hostonly" tag? Normally we'd filter passing/failing platforms a little differently...

Reply:
The benchmark suites are not compiled for all platforms as they are defined in separated  `.cmake` files in `benchmarks/` I can add test flags to define platforms and set the ctest to filter the label there.

Reply:
Add the label to bypass certain platforms. `XFAIL` now is specifically for the correctness failure.

Reply:
Thanks for the suggestion! Updated the logic to use `get_platform` function to check `UNSUPPORTED_PLATFORMS` and `XFAIL`. I made the default of `_PLATFORM` to be `"${CMAKE_SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}"`; however, android.toolchain.cmake doesn't use those native variables so I need to enumerate that condition. :(